File: Api/Entities/Producto.cs


using System.ComponentModel.DataAnnotations;
using Api.Entities;

namespace Api.Data
{
    public class Producto
    {
        [Key]
        public required string Id { get; set; }
        public required string Nombre { get; set; }
        public float Precio { get; set; }
        public string Departamento { get; set; }
        public int Ganancias { get; set; }

        public List<Foto> Fotos { get;set;}
        public List<Electrodomestico> Electrodomesticos { get;set;}
        public List<Informatica> Informaticas { get;set;}
        public List<Stock> Stocks { get;set;}

    }
}

File: client/src/app/_models/Producto.ts
import { Foto } from "./Foto"
import { Stock } from "./Stock"

export interface Producto {
    id: string,
    nombre: String,
    precio: number,
    departamento: string
    tipo: string
    descripcion: string
    marca: string
    consumo: string
    localizacion: string
    valor: string
    fotoUrl: string
    fotos: Foto[]
    stocks: number
    cantidadVendidos : number
    cantidadVenta : number
    ganancias: number
}
